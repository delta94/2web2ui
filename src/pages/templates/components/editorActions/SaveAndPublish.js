import React from 'react';
import { CheckCircleOutline } from '@sparkpost/matchbox-icons';
import { Button } from 'src/components/matchbox';
import { ButtonLink } from 'src/components/links';

const SaveAndPublish = props => {
  const { onClick, className, children } = props;

  return (
    <div className={className}>
      {children && (
        <Button variant="secondary" onClick={onClick} data-id="action-save-and-publish">
          {children}
        </Button>
      )}

      {!children && (
        <ButtonLink onClick={onClick} data-id="action-save-and-publish">
          <CheckCircleOutline />

          <span>Save and Publish</span>
        </ButtonLink>
      )}
    </div>
  );
};

export default SaveAndPublish;
