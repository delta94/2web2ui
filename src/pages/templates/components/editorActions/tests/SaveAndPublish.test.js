import React from 'react';
import { shallow } from 'enzyme';
import SaveAndPublish from '../SaveAndPublish';
import useEditorContext from '../../../hooks/useEditorContext';

jest.mock('../../../hooks/useEditorContext');

describe('SaveAndPublish', () => {
  const subject = (editorState, children = null, props) => {
    useEditorContext.mockReturnValue({
      draft: { id: 'foo' },
      history: { push: jest.fn() },
      ...editorState,
    });

    return shallow(
      <SaveAndPublish className="Foo" {...props}>
        {children}
      </SaveAndPublish>,
    );
  };

  it('renders SaveAndPublish action', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('renders SaveAndPublish action with provided children', () => {
    expect(subject(null, <span>Click here</span>).find('Button')).toExist();
  });

  it('Renders and icon and default content when no children are supplied', () => {
    const wrapper = shallow(<SaveAndPublish />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders confirmation modal upon clicking', () => {
    const mockFn = jest.fn();
    const wrapper = subject(undefined, null, { onClick: mockFn });

    wrapper.find('ButtonLink').simulate('click');

    expect(mockFn).toHaveBeenCalled();
  });
});
