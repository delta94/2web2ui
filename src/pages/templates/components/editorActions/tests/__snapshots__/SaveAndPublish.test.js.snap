// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveAndPublish Renders and icon and default content when no children are supplied 1`] = `
<div>
  <ButtonLink
    data-id="action-save-and-publish"
  >
    <CheckCircleOutline />
    <span>
      Save and Publish
    </span>
  </ButtonLink>
</div>
`;

exports[`SaveAndPublish renders SaveAndPublish action 1`] = `
<div
  className="Foo"
>
  <ButtonLink
    data-id="action-save-and-publish"
  >
    <CheckCircleOutline />
    <span>
      Save and Publish
    </span>
  </ButtonLink>
</div>
`;
